content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM.class/__methods__/retry_placement.rb

```diff
@@ -0,0 +1,29 @@
+#
+# Description: This method updates the provisioning status and retries
+# from the Placement step.
+# Required inputs: status
+#
+
+prov = $evm.root['miq_provision']
+unless prov
+  $evm.log(:error, "miq_provision object not provided")
+  exit(MIQ_STOP)
+end
+status = $evm.inputs['status']
```

Comment:
@mansam 
@gmcculloug Correct me if I am wrong.
There are 2 state machines here, 1 internal inside the app/models and the other one in Automate.

Internal State Machine:
The start_clone_task is an asynchronous process that talks to the provider to get the provisioning started.
Then the poll_clone_complete waits with a retry and then signals the vmdb in database after it has done an ems_refresh.

https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/app/models/manageiq/providers/vmware/infra_manager/provision/state_machine.rb#L28



The external state machine define in Automate is in the check_provisioned loop waiting for the machine to show up  in our database my checking the status of the task.

https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb#L7

So I am guessing we would have to depend on the provisioning task to provide more errors from the cloning process, I don't think we have an explicit column that stores the error code from the provider. So I think we would have to parse the tasks error message and see if we can glean something from there to trigger a retry on placement.